src/main/java/org/opensearch/neuralsearch/util/NeuralSearchClusterUtil.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.util;
+
+import java.util.Locale;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Log4j2
+public class NeuralSearchClusterUtil {
```

Comment:
do we really need this class cannot we directly get it from any OpenSearch class?

Reply:
I would suggest let's keep this class as it provides abstraction from underlying opensearch cluster, also if i have to remove this then I have initialize cluster service variable in the create components and then have take it all the way to neural query builder.

Reply:
Because Version data comes from clusterService.state().nodes().getMinNodeVersion()